Routers/Routers/Graph.cs

```diff
@@ -0,0 +1,118 @@
+namespace Routers;
+
+using UtilityExceptions;
+
+/// <summary>
+/// Graph class with adjacency matrix as a property and dfs, connection checking and parsing methods
+/// </summary>
+public class Graph : IGraph
+{
+    private int _arrayLength = 10;
+
+    public int MaxNodeNumber { get; private set; }
+
+    public int[,] Matrix { get; private set; }
+
+    public Dictionary<(int, int), int> Edges { get; set; }
+
+    public Graph(string inputFilePath)
+    {
+        Matrix = new int[_arrayLength, _arrayLength];
+        Edges = new Dictionary<(int, int), int>();
+        Parse(inputFilePath);
+    }
+
+    private void Parse(string inputFilePath)
+    {
+        using StreamReader reader = new StreamReader(inputFilePath);
```

Comment:
```suggestion
        using var reader = new StreamReader(inputFilePath);
```
Это как заикание, "using StreamReader-StreamReader" :)